src/tasks.ts

```diff
@@ -0,0 +1,63 @@
+import { Either, isLeft } from "fp-ts/lib/Either";
+import { Task } from "fp-ts/lib/Task";
+import { Millisecond } from "./units";
+
+export type TransientError = "transient";
+export const TransientError: TransientError = "transient";
+
+export type MaxRetries = "max-retries";
+export const MaxRetries: MaxRetries = "max-retries";
+
+/**
+ * A Task that can be retried when it fails with a transient error
+ */
+export type RetriableTask<E, T> = E extends TransientError
+  ? never
+  : Task<Either<E | TransientError, T>>;
```

Comment:
you may use a TaskEither

Reply:
I actually started by using `TaskEither` but then I decided to do a first implementation with the types I knew already and then later refactor as an optimization

Reply:
yes that's because AFAIK there's no `fromTask()` method in TaskEither cc @gcanti

Reply:
I guess `fromTask` is `right`. Some lifting functions:

- `taskEither.of: <L, A>(a: A) => TaskEither<L, A>` (*)
- `right: <L, A>(fa: Task<A>) => TaskEither<L, A>`
- `left: <L, A>(fa: Task<L>) => TaskEither<L, A>`

(*) where `taskEither` comes from `import { taskEither } from 'fp-ts/lib/TaskEither'`

(see https://github.com/gcanti/fp-ts/blob/master/docs/TaskEither.md)